pkg/lakom/resolvetag/admission.go

```diff
@@ -193,7 +202,15 @@ func (h *handler) handlePod(ctx context.Context, p *corev1.Pod, logger logr.Logg
 func (h *handler) handleContainer(ctx context.Context, image string, kcr utils.KeyChainReader, logger logr.Logger) (string, error) {
 	logger = logger.WithValues("originalImage", image)
 
-	imageRef, err := name.ParseReference(image)
+        var (
+            imageRef name.Reference
+            err error
+        )
+        if h.allowInsecureRegistries {
+            imageRef, err = name.ParseReference(image, name.Insecure)
+        } else {
+            imageRef, err = name.ParseReference(image)
+        }
```

Comment:
I would propose something like 
```go
	opts := []name.Option{}
	if h.allowInsecureRegistries {
		opts = append(opts, name.Insecure)
	}
	imageRef, err := name.ParseReference(image, opts...)
```

This way there is no else statement and if more options needs to be enabled, it will be straight forward.

Reply:
Thanks that's much better